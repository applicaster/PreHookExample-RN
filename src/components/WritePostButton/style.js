import { StyleSheet } from 'react-native';

const buttonSize = 60;
export const styles = StyleSheet.create({
  writePostButtonContainer: {
    position: 'absolute',
    bottom: 15,
    right: 10,
    height: buttonSize,
    width: buttonSize,
  },
  writePostButton: {
    height: buttonSize,
    width: buttonSize,
  },
  writePostPencil: {
    height: 35,
    width: 30,
    bottom: 15,
    right: 14,
    position: 'absolute',
  },
});
