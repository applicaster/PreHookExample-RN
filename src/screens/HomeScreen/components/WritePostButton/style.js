import { StyleSheet } from 'react-native';

const buttonSize = 54;
const pencilSize = 17;
export const styles = StyleSheet.create({
  writePostButtonContainer: {
    position: 'absolute',
    bottom: 10,
    right: 10,
    height: buttonSize,
    width: buttonSize,
  },
  writePostButton: {
    height: buttonSize,
    width: buttonSize,
    shadowColor: '#000000',
    shadowOffset: { width: 1, height: 1 },
    shadowOpacity: 0.5,
    shadowRadius: 3,
  },
  writePostPencil: {
    height: pencilSize,
    width: pencilSize,
    bottom: 20,
    right: 18,
    position: 'absolute',
  },
});
