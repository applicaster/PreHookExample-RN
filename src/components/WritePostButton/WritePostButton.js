import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  TouchableOpacity,
  View,
} from 'react-native';
import { styles } from './style';

const WRITE_POST_BUTTON = 'feed_rn_write_post_button';
const PENCIL_IMAGE = 'feed_rn_pencil';

class WritePostButton extends Component {
  render() {
    const { mainColor } = this.context;
    const { openWritePostModal } = this.props;
    const buttonTintColor = { tintColor: mainColor };

    return (
      <TouchableOpacity onPress={() => openWritePostModal({ modal: 'WritePostModal' })}>
        <View style={styles.writePostButtonContainer}>
          <Image style={[styles.writePostButton, buttonTintColor]} source={{ uri: WRITE_POST_BUTTON }} />
          <Image style={[styles.writePostPencil]} source={{ uri: PENCIL_IMAGE }} />
        </View>
      </TouchableOpacity>
    );
  }
}

WritePostButton.contextTypes = {
  mainColor: PropTypes.string,
};

WritePostButton.propTypes = {
  openWritePostModal: PropTypes.func,
};

export default WritePostButton;
