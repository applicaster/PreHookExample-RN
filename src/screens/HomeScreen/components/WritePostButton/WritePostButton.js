import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  TouchableOpacity,
} from 'react-native';
import { Image } from 'react-native-animatable';
import { sendAnalyticEvent } from 'react-native-zapp-bridge';
import { WRITE_POST_BUTTON_CLICKED } from '../../../../constants/analyticEvents';
import { styles } from './style';

const WRITE_POST_BUTTON = 'feed_rn_write_post_button';
const PENCIL_IMAGE = 'feed_rn_pencil';

class WritePostButton extends Component {
  constructor(props) {
    super(props);
    this.onPress = this.onPress.bind(this);
  }

  onPress() {
    const { openWritePostModal } = this.props;
    
    openWritePostModal({ modal: 'WritePostModal' });
    sendAnalyticEvent(WRITE_POST_BUTTON_CLICKED, {}).then().catch();
  }

  render() {
    const { mainColor, backgroundColor } = this.context;
    const { isCardActive, isFacebookAvailable, isSocialPostingEnabled, isTwitterAvailable } = this.props;
    if (isCardActive || !isSocialPostingEnabled || (!isFacebookAvailable && !isTwitterAvailable)) return null;
    
    const buttonTintColor = { tintColor: mainColor };
    const pencilTintColor = { tintColor: backgroundColor };

    const ANIMATION_TYPE = 'fadeInUp';
    const ANIMATION_DURATION = 400;
    const ANIMATION_DELAY = 650;
    return (
      <TouchableOpacity activeOpacity={0.85} style={styles.writePostButtonContainer} onPress={this.onPress}>
        <Image
          animation={ANIMATION_TYPE}
          delay={ANIMATION_DELAY}
          duration={ANIMATION_DURATION}
          source={{ uri: WRITE_POST_BUTTON }}
          style={[styles.writePostButton, buttonTintColor]}
        />
        <Image
          animation={ANIMATION_TYPE}
          delay={ANIMATION_DELAY}
          duration={ANIMATION_DURATION}
          source={{ uri: PENCIL_IMAGE }}
          style={[styles.writePostPencil, pencilTintColor]}
        />
      </TouchableOpacity>
    );
  }
}

WritePostButton.contextTypes = {
  mainColor: PropTypes.string,
  backgroundColor: PropTypes.string,
};

WritePostButton.propTypes = {
  isCardActive: PropTypes.bool.isRequired,
  isFacebookAvailable: PropTypes.bool.isRequired,
  isTwitterAvailable: PropTypes.bool.isRequired,
  isSocialPostingEnabled: PropTypes.bool.isRequired,
  openWritePostModal: PropTypes.func,
};

WritePostButton.defaultProps = {
  isCardActive: false,
  isFacebookAvailable: false,
  isTwitterAvailable: false,
  isSocialPostingEnabled: true,
};

export default WritePostButton;
