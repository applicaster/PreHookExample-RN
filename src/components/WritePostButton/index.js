import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleModal } from '../../actions';
import WritePostButton from './WritePostButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  openWritePostModal: toggleModal,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(WritePostButton);
