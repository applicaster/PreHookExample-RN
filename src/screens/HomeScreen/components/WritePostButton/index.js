import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleModal } from '../../../../actions';
import WritePostButton from './WritePostButton';
import {
  isFacebookAvailable,
  isTwitterAvailable,
} from '../../../../selectors';

const mapStateToProps = state => ({
  isFacebookAvailable: isFacebookAvailable(state),
  isTwitterAvailable: isTwitterAvailable(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  openWritePostModal: toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(WritePostButton);
