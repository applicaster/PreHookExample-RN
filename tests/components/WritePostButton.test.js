import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WritePostButton from '../../src/components/WritePostButton/WritePostButton';

jest.mock('@applicaster/react-native-zapp-bridge', () => 'NativeZappBridge');

describe('WritePostButton', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <WritePostButton />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
