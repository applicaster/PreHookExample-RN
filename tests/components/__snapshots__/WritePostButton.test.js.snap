// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritePostButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "bottom": 15,
        "height": 60,
        "position": "absolute",
        "right": 10,
        "width": 60,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "feed_rn_write_post_button",
        }
      }
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          Object {
            "tintColor": undefined,
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "uri": "feed_rn_pencil",
        }
      }
      style={
        Array [
          Object {
            "bottom": 15,
            "height": 35,
            "position": "absolute",
            "right": 14,
            "width": 30,
          },
        ]
      }
    />
  </View>
</View>
`;
